### ISSUE TITLE
Can't connect to Azure MySQL instance when "enforce SSL connection" is on

### ISSUE THREAD

**Issue by ADBurpitt** [milvus-io/milvus#5330]
**Describe the bug**

I am attempting to use an Azure managed MySQL instance with milvus, when I enable the option "enforce SSL connection" the following error occurs:

`Error: failed to open database: Can't construct MySQL connection.`

With the option disabled the connection is ok.

Is there any way around this? I am able to connect to the DB with a normal python client so I don't think the issue is on that side.

I would rather not have to disable SSL, has anyone else had this problem or is there something I've missed as far as how to get this working?

**Expected behavior**

Milvus should be able to connect to MySQL using TLS.

**Method of installation**
- [x] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Software conditions
  - Running in AKS, deployed via the milvus helm chart
- Milvus version
  - 1.1

**Comment by yhmo** [comment]
I just search this issue on google. This [link](https://stackoverflow.com/questions/44035588/why-does-connection-to-my-mysql-server-in-azure-fail-if-my-app-does-not-have-ssl) give two options: 
1. disable  Enforce SSL
2. download a certificate from azure, and use connect parameter to apply the certificate, "-p --ssl-ca=/opt/ssl/BaltimoreCyberTrustRoot.crt.pem"

For the option 2, currently milvus use regex expression to parse the uri from config file, it not able to parse the additional parameter. I think this is an improvement.
The meta uri parse code is [here](https://github.com/milvus-io/milvus/blob/1.1/core/src/db/Utils.cpp#L261) 
The mysql connection is created [here](https://github.com/milvus-io/milvus/blob/1.1/core/src/db/meta/MySQLConnectionPool.cpp#L63) 

So, to support ssl, we need to change the code of uri parser and connection builder.
Milvus use mysqlpp as thirdparty to interact with mysql, When we create a mysql connection, we need to add an option to input ssl parameter like this:
```
    mysqlpp::Connection* conn = new mysqlpp::Connection();
    conn->set_option(new mysqlpp::SslOption(key, cert, ca, capath, cipher));
```

**Comment by RyanWei** [comment]
Seems this an easy fix, can we have this fixed with priority? Data security is very critical on public cloud.

**Comment by xiaofan-luan** [comment]
> Seems this is an easy fix, can we have this fixed with priority? Data security is very critical on public cloud.

@yhmo  can we add SslOption in 1.1.2?  It seems azure SQL force SSL and option1 will not work.
@ADBurpitt  it's a simple fix, yhmo has already point out the line of code to add, you may want to wait for next release or modified current code yourself, thanks.

**Comment by yhmo** [comment]
- MySQL ssl connection levels:
```
CREATE USER ...
[REQUIRE {NONE | tls_option [[AND] tls_option] ...}]
...
tls_option: {
 SSL
 | X509
 | CIPHER 'cipher'
 | ISSUER 'issuer'
 | SUBJECT 'subject'
}
```
"SSL" only require a certification generated by private RSA key
"X509" require a certification generated by private RSA key, and client RSA key and client certification

Looks like the Azure MySQL ssl tls_option is "SSL".

- To pass certification file path into Milvus, will change the server_config.yaml to:
```
general:
  timezone: UTC+8
  meta_uri: mysql://xxx:xxxxxx@127.0.0.1:3306/milvus
  meta_ssl_ca:
  meta_ssl_key:
  meta_ssl_cert:
```
"meta_ssl_ca" is the file path of certification generated by private RSA key. For Azure MySQL, download from this link: https://cacerts.digicert.com/DigiCertGlobalRootG2.crt.pem
"meta_ssl_key" and "meta_ssl_cert" are for tls_option "X509", the file path of client RSA key and client certification

**Comment by yhmo** [comment]
The #5330 was fixed on branch 1.1, the official docker image of 1.1.2 is not yet published, you can download a release-candidate image:
docker pull milvusdb/milvus-poc:1.1.2-cpu-rc
docker pull milvusdb/milvus-poc:1.1.2-gpu-rc
Follow these steps:
Start a MySQL server on Azure, create a database named "milvus"
To connect Azure MySQL server, you need to download a pem file from Azure online documentation.
 3.  Download milvus v1.x configuration file
wget https://raw.githubusercontent.com/milvus-io/milvus/v1.1.1/core/conf/demo/server_config.yaml
 4.  Modify the configuration file, add "meta_ssl_ca" to tell milvus the pem file path:
general:
  timezone: UTC+8
  meta_uri: mysql://xxx:xxxxxx@xxxxxxxxxxxxxx:3306/milvus
  meta_ssl_ca: [path to the pem file]
 5. Put the configuration file into /home/$USER/milvus/conf. Start Milvus server using the configuration file:
sudo docker run -d --name milvus_cpu_1.1.2 \
-p 19530:19530 \
-p 19121:19121 \
-v /home/$USER/milvus/db:/var/lib/milvus/db \
-v /home/$USER/milvus/conf:/var/lib/milvus/conf \
-v /home/$USER/milvus/logs:/var/lib/milvus/logs \
-v /home/$USER/milvus/wal:/var/lib/milvus/wal \
milvusdb/milvus:milvus-poc:1.1.2-cpu-rc****
